src/code-editor/index.tsx

```diff
@@ -41,7 +41,7 @@ import styles from './styles.css.js';
 
 export { CodeEditorProps };
 
-export default function CodeEditor(props: CodeEditorProps) {
+const CodeEditor = forwardRef((props: CodeEditorProps, ref: React.Ref<CodeEditorProps.Ref>) => {
```

Comment:
This code gives external consumers access to the entire `<div>` node and this is not what we want.

Please check how other components implement focus feature (hint, there is `useFowardFocus` hook)

Reply:
good catch, thanks boris!